### PR TITLE
2454 Add documentation on how to work with xml

### PR DESCRIPTION
**Pull request by rafalambrozewicz**
* Added documentation on how to integrate with scalaxb tool for XML serialization/deserialization.
* In "OpenAPI Specification Extensions" added an example on how to add doc extension on the schema.

### CODE REVIEW

File: doc/docs/openapi.md [hunk 1]
```diff
@@ -270,6 +270,9 @@ case class FruitAmount(fruit: String, amount: Int)
 
 case class MyExtension(string: String, int: Int)
 
+implicit val fruitAmountSchemaWithMyExtension: Schema[FruitAmount] = implicitly[Derived[Schema[FruitAmount]]].value
```
Comment (adamw) [inline doc/docs/openapi.md:273]: or maybe simpler: `Schema.derived[FruitAmount].docsExt(...)`

File: doc/endpoint/xml.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Working with XML
+
+Enabling support for XML is a matter of implementing proper `XmlCodec[T]` and providing it in scope.
+This would enable encoding objects to XML strings, and decoding XML strings to objects.
```
Comment (adamw) [inline doc/endpoint/xml.md:4]: [minor] This enables ...

File: doc/endpoint/xml.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# Working with XML
+
+Enabling support for XML is a matter of implementing proper `XmlCodec[T]` and providing it in scope.
```
Comment (adamw) [inline doc/endpoint/xml.md:3]: maybe link to codec.md?

File: doc/endpoint/xml.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  Note, that implementing `XmlCodec[T]` would require deriving not only XML library encoders/decoders, 
+  but also tapir related `Schema[T]`. These are completely separate - any customization e.g. for field
+  naming or inheritance strategies must be done separately for both derivations.
+  For more details see sections on [schema derivation](schemas.md) and on supporting [custom types](customtypes.md) in general.
+```
+
+## Scalaxb
+
+If you possess the XML Schema definition file (`.xsd` file) consider using the scalaxb tool,
+which would generate needed models and serialization/deserialization logic.
```
Comment (adamw) [inline doc/endpoint/xml.md:19]: again I would just say "which generates ..."

File: doc/endpoint/xml.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+It also introduces `xmlBody` helper method, which allows you to easily express, that the declared endpoint consumes or returns XML.
+
+
+Next to this trait, you might want to introduce `xml` package object to simplify imports.
+```scala
+package object xml extends TapirXmlScalaxb
+```
+
+From now on, XML serialization/deserialization would work for all classes generated from `.xsd` file as long as `XMLFormat`
+and `Schema` for the type in the question, `XmlElementLabel` for the top XML node would be implicitly provided in the scope.
```
Comment (adamw) [inline doc/endpoint/xml.md:64]: I think this could be a list, plus I would elaborate a bit on what's the label for

**Review by adamw — APPROVED** [review]

File: doc/docs/openapi.md (hunk 1 above)
Comment (rafalambrozewicz) [reply doc/docs/openapi.md]: Great suggestion, updated, thanks! 👍

File: doc/endpoint/xml.md (hunk 2 above)
Comment (rafalambrozewicz) [reply doc/endpoint/xml.md]: updated

File: doc/endpoint/xml.md (hunk 3 above)
Comment (rafalambrozewicz) [reply doc/endpoint/xml.md]: updated

File: doc/endpoint/xml.md (hunk 4 above)
Comment (rafalambrozewicz) [reply doc/endpoint/xml.md]: updated

File: doc/endpoint/xml.md (hunk 5 above)
Comment (rafalambrozewicz) [reply doc/endpoint/xml.md]: updated

File: doc/endpoint/xml.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+If the generation of OpenAPI documentation is required, consider adding OpenAPI doc extension on schema providing XML
+namespace as described in the "Prefixes and Namespaces" section at [OpenAPI documentation regarding handling XML](https://swagger.io/docs/specification/data-models/representing-xml/).
+This would add `xmlns` property to example request/responses at swagger, which is required by scalaxb to properly deserialize XML.
+For more information on adding OpenAPI doc extension in tapir refer to [documentation](../docs/openapi.md#openapi-specification-extensions).
+
+Adding xml namespace doc extension to `Outer`'s `Schema` example:
+```scala
+case class XmlNamespace(namespace: String)
+implicit val outerSchemaWithXmlNamspace: Schema[Outer] = implicitly[Derived[Schema[Outer]]].value
```
[the remark this answers is not in the document]
Comment (rafalambrozewicz) [reply doc/endpoint/xml.md]: thanks, updated :-)

**Review by geminicaprograms — APPROVED** [review]

**Pull request merged**